src/gui/tray/Window.qml

```diff
@@ -624,56 +616,156 @@ Window {
+
+                            textBorderColor: Style.ncBlue
+
+                            textBgColor: "transparent"
+                            textBgColorHovered: Style.ncBlue
+
+                            onClicked: activityModel.triggerAction(activityItem.itemIndex, actionIndex)
+
+                            tooltipText: !primary ? activityItem.links[actionIndex].label : ""
+
+                            Layout.minimumWidth: primary ? 80 : -1
+                            Layout.minimumHeight: parent.height
+
+                            Layout.preferredWidth: primary ? -1 : parent.height
+                        }
 
-                    Layout.preferredWidth: (objectType == "chat" || objectType == "call") ? parent.height : 0
-                    Layout.preferredHeight: parent.height
-                    Layout.alignment: Qt.AlignRight
-                    flat: true
-                    hoverEnabled: true
-                    visible: (objectType == "chat" || objectType == "call") ? true : false
-                    display: AbstractButton.IconOnly
-                    icon.source: "qrc:///client/theme/reply.svg"
-                    icon.color: "transparent"
-                    background: Rectangle {
-                        color: parent.hovered ? Style.lightHover : "transparent"
                     }
-                    ToolTip.visible: hovered
-                    ToolTip.delay: 1000
-                    ToolTip.text: qsTr("Open Talk")
-                    onClicked: Qt.openUrlExternally(link)
 
-                    Accessible.role: Accessible.Button
-                    Accessible.name: qsTr("Open Talk %1").arg(link)
-                    Accessible.onPressAction: replyButton.clicked()
+                    Button {
+                        id: moreActionsButton
+
+                        Layout.preferredWidth: parent.height
+                        Layout.preferredHeight: parent.height
+                        Layout.alignment: Qt.AlignRight
+
+                        flat: true
+                        hoverEnabled: true
+                        visible: activityItem.links.length > 2
+                        display: AbstractButton.IconOnly
+                        icon.source: "qrc:///client/theme/more.svg"
+                        icon.color: "transparent"
+                        background: Rectangle {
+                            color: parent.hovered ? Style.lightHover : "transparent"
+                        }
+                        ToolTip.visible: hovered
+                        ToolTip.delay: 1000
+                        ToolTip.text: qsTr("Show all actions")
+                        onClicked:  moreActionsButtonContextMenu.popup();
+
+                        Connections {
+                            target: trayWindow
+                            onActiveChanged: {
+                                if (!trayWindow.active && moreActionsButtonContextMenu.opened) {

```

Comment:
I wouldn't check on `.opened` and just close. If it's not opened it'll be a noop anyway.